tools/value-generation/gen.py

```diff
@@ -153,9 +159,17 @@ def purge_unset(_dict):
 
     if 'properties' in value:
       purge_unset(value['properties'])
+      if len(value['properties']) == 0:
```

Comment:
Isn't this post purge? I think I was getting empty leaves as all nested were removed.

It may be that indeed I should have moved the check up, I'll verify.

Reply:
Yeah I'm getting various extra un-purged nodes that only have empty leaves:
```
   # -- Backup defines whether the Operator should manage automated backups, and
   # how to lookup backup resources.
   backup:
+    # -- The Backup Image to run on backup pods
     image: couchbase/operator-backup:1.0.0
+    # -- ImagePullSecrets allow you to use an image from private repositories
+    # and non-dockerhub ones.
+    imagePullSecrets:
+      # -- Name of the referent. More info:
+      # https://kubernetes.io/docs/concepts/overview/working-with-
+      # objects/names/#names TODO: Add other useful fields. apiVersion, kind,
+      # uid?
+      name:
```

Reply:
The reason is inside purge you pop off the keys and can make it empty I think.